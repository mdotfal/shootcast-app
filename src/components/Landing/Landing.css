/* Body Styles */

body {
  font-family: 'Noto Sans JP', sans-serif;
  text-align: center;
}

/* Input / Button Styles */

input[ type=text ],
input[ type=password ],
button,
button[ type=submit ] {
  width: 100px;
  margin: 10px;
}

input#inputList,
input#search,
select#listId,
input#username,
input#password {
  width: 120px;
}

input#inputList,
input#search {
  width: 120px;
  -webkit-transition: width 0.5s ease-in-out;
  transition: width 0.5s ease-in-out;
}

input#inputList:focus,
input#search:focus {
  width: 70%;
}

input#inputList,
input#search,
input#username,
input#password,
input#password-repeat {
  border: none;
  border-bottom: 2px solid #5e351b;
}

/* Button Styles */

.button,
button,
button[ type=submit ] {
  margin: 10px;
  font-size: 14px;
  background-color: #F6EDE7;
  box-shadow: 0 1px 2px 0;
  border-radius: 25px;
  padding: 5px;
  text-decoration: none;
  color: black;
}

/* Landing Page Styles */

.landing-page {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 90vw;
  height: 100vh;
}

.landing-page h1 {
  padding: 15px;
}

.landing-container {
  display: flex;
  flex-direction: column;
  align-items: center;
}

fieldset {
  max-width: 800px;
  margin: 20px;
}
