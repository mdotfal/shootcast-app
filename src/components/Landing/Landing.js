import React, { Component } from 'react';
import { Link, Redirect } from 'react-router-dom';
import AppContext from '../../AppContext';
import './Landing.css';
class Landing extends Component {

  static contextType = AppContext;

  state = {
    username: '',
    password: '',
    isError: false,
    authedUser: false,
  }
  
  onUserChange = e => {
    this.setState({
      username: e.target.value
    })
  }

  onPassChange = e => {
    this.setState({
      password: e.target.value
    })
  }

  onClick = e => {
    this.setState({
      authedUser: true
    })
  }

  onSubmit = e => {
    e.preventDefault();
    const result = this.props.handleLogin( this.state.username, this.state.password );
    if( !result ) {
      this.setState({
        isError: true,
      })
    }
  }

  render() {
  
    if( this.props.authedUser ) {
      return (
        <Redirect to='/home' />
      );
    };

    return (
      <div className="landing-page">
        <h1>Welcome to the ShootCast App</h1>
        <p>In a world of varying climates and back-to-back creative shoots, a way to keep in the know of current weather situations is a must.</p>
        <br />
        <p>ShootCast is a mobile app that will help you do so and create custom lists of cities to see the weather conditions with the click of a mouse.</p>
        <br />
        <p>Find out the current weather forecast and plan your shoot correctly.</p>

        { this.state.isError && <div style={{ color: 'red' }}> Username/Password incorrect </div> }
        
        <form onSubmit={ this.onSubmit }>
          <fieldset>
            <div className="landing-container">
              <label htmlFor="username"><b>Username</b></label>
              <input type="text" placeholder="Enter Username" name="username" id="username" onChange={ this.onUserChange } required />
          
              <label htmlFor="password"><b>Password</b></label>
              <input type="password" placeholder="Enter Password" name="password" id="password" onChange={ this.onPassChange} required />

              <div className="landing-buttons">
                <button 
                  type="submit"
                >Login</button>
                <Link to='/home'>
                  <button onClick={ () => this.context.onGuestLogin() }>Guest Login</button>
                </Link>
              </div>
            </div>
          </fieldset>
        </form>
        or
        <br />
        <Link to="/registration">
          <button>Register</button>
        </Link>
        
      </div>
    );
    
  }
}

export default Landing;