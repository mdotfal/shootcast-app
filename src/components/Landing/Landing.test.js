/* eslint-disable jest/valid-describe */
import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom/cjs/react-router-dom.min';
import Landing from './Landing';

describe( 'Landing Component', () => {
  test( 'should render', () => {
    const component = renderer
      .create(
        <BrowserRouter>
          <Landing />
        </BrowserRouter>
      ).toJSON();
    expect( component ).toMatchSnapshot();
  })
})