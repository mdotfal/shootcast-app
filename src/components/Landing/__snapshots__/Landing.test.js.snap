// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Component should render 1`] = `
<div
  className="landing-page"
>
  <h1>
    Welcome to the ShootCast App
  </h1>
  <p>
    In a world of varying climates and back-to-back creative shoots, a way to keep in the know of current weather situations is a must.
  </p>
  <br />
  <p>
    ShootCast is a mobile app that will help you do so and create custom lists of cities to see the weather conditions with the click of a mouse.
  </p>
  <br />
  <p>
    Find out the current weather forecast and plan your shoot correctly.
  </p>
  <form
    onSubmit={[Function]}
  >
    <fieldset>
      <div
        className="landing-container"
      >
        <label
          htmlFor="username"
        >
          <b>
            Username
          </b>
        </label>
        <input
          id="username"
          name="username"
          onChange={[Function]}
          placeholder="Enter Username"
          required={true}
          type="text"
        />
        <label
          htmlFor="password"
        >
          <b>
            Password
          </b>
        </label>
        <input
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Enter Password"
          required={true}
          type="password"
        />
        <div
          className="landing-buttons"
        >
          <button
            type="submit"
          >
            Login
          </button>
          <a
            href="/home"
            onClick={[Function]}
          >
            <button
              onClick={[Function]}
            >
              Guest Login
            </button>
          </a>
        </div>
      </div>
    </fieldset>
  </form>
  or
  <br />
  <a
    href="/registration"
    onClick={[Function]}
  >
    <button>
      Register
    </button>
  </a>
</div>
`;
